runtime/src/main/java/org/corfudb/runtime/view/metadata/Checkpoint.java

```diff
@@ -0,0 +1,14 @@
+package org.corfudb.runtime.view.metadata;
+
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * This class defines metadata associated with a checkpoint
+ */
+public class Checkpoint {
+    private UUID checkpointStream;
+    private long startAddress;
+    private long endAddress;
+    List<Long> checkpointRecords;
```

Comment:
For the recovery loading of the maps in one pass I would also like to have all the addresses.  As Maithem mentioned, I don't have to follow backpointers from the end address, which is nice, since I read forward.